raiden/tests/conftest.py

```diff
@@ -65,6 +66,26 @@ def pytest_addoption(parser):
     )
 
 
+@pytest.fixture(autouse=True, scope="session")
+def check_geth_version_for_tests():
+    geth_version_string, _ = subprocess.Popen(
+        ["geth", "version"], stdout=subprocess.PIPE, stderr=subprocess.PIPE
+    ).communicate()
+    match = re.search("Version: (.*)-", geth_version_string.decode())
+    if match:
+        # Silently ignore unexpected output format from geth version
+        if len(match.groups()) == 1:
+            our_version = parse_version(match.groups()[0])
+            max_supported_version = parse_version(HIGHEST_SUPPORTED_GETH_VERSION)
+
+            if our_version > max_supported_version:
```

Comment:
This would also be very useful for a user, we should also add this to the startup of the app at `cli.py`

Reply:
Good point. Will do. Any idea how to sync the constant of the version with the one in the circleci config yaml? This is my big question when writting this patch.

Reply:
This [one](https://github.com/raiden-network/raiden/blob/cfc4996e67d9104b3a1c2ea51ca6d782f3809678/.circleci/config.yml#L44)?

There are two things here:

- The constant in the config.yaml is used rename the geth binary because of how caches work.
- The geth binary downloaded in the build must be smaller than the version that we support (which this PR introduces the check for)

For the first case, removing the constant would fix synchronization problems, this would just need a bit of additional shell scripting. The benefit here would be a guarantee that the cache is correct.

For the second case, I guess we can just rely on this check? If someone just upgrades the Geth version, say to 1.9.0, without upgrading the constant then the build will fail. This should be sufficient to force the PR author to upgrade the constant too and make sure the build is green.

Reply:
Yep second one is what I meant. I guess just relying on the build failing is good enough and serves the purpose of what we need.